src/rpc/common/impl/Processors.h

```diff
@@ -31,21 +31,61 @@ struct DefaultProcessor final
     operator()(HandlerType const& handler, boost::json::value const& value)
         const
     {
-        using boost::json::value_from;
-        using boost::json::value_to;
+        // can not do the check in one if statement
+        // first check the nested type exist
+        // then check has a process function called by this type
+        constexpr bool hasNonVoidInputField =
+            requires(const HandlerType& t) { typename HandlerType::Input; };
+        if constexpr (hasNonVoidInputField)
```

Comment:
If you split into smaller concepts in Concepts.h you can probably leverage it here?